core/src/main/clojure/xtdb/expression/temporal.clj

```diff
@@ -1440,6 +1440,49 @@
                       "MILLISECOND" `(->period-duration ~period (.truncatedTo ~duration ChronoUnit/MILLIS))
                       "MICROSECOND" `(->period-duration ~period (.truncatedTo ~duration ChronoUnit/MICROS)))))})
 
+(defn ->mdn-interval-between [^LocalDateTime end-dt ^LocalDateTime start-dt]
+  (let [period-between (Period/between (.toLocalDate start-dt) (.toLocalDate end-dt))
+        period-days (.getDays period-between)
+        duration-between (Duration/between start-dt end-dt)
+        start-lt (.toLocalTime start-dt)
+        end-lt (.toLocalTime end-dt)
+        period-day-adjustment (cond
```

Comment:
mind walking me through the logic here when you get a mo?

(I'm sure it's fine, it's just not intuitive to me!)